### PR TITLE
Update docsite with some notes on which APIs are stabilised and which are not.

### PR DESCRIPTION
**Pull request by mikehearn**
Also move coding style to 'other' where it belongs.

### CODE REVIEW

**Comment by mnesbit** [comment]
I think we need to state that the core DB schemas are subject to change without notice as are serialization formats, but looks consistent with the rules we have been applying.

File: docs/source/api-index.rst [hunk 1]
```diff
@@ -16,11 +16,34 @@ This section describes the APIs that are available for the development of CorDap
...
+
+As of Corda 1.0, the following modules export public API that we promise to maintain backwards compatibility for,
+unless an incompatible change is required for security reasons:
+
+* core
+* client-rpc
+* client-jackson
+* node-driver
+* node-api
+* The Gradle plugins: cordformation, quasar-utils
```
Comment (shamsasari) [inline docs/source/api-index.rst]: You may want to confirm with @Clintonio on this

File: docs/source/api-index.rst [hunk 2]
```diff
@@ -16,11 +16,34 @@ This section describes the APIs that are available for the development of CorDap
...
+--------------------------------------
+
+As of Corda 1.0, the following modules export public API that we promise to maintain backwards compatibility for,
+unless an incompatible change is required for security reasons:
+
+* core
+* client-rpc
+* client-jackson
+* node-driver
+* node-api
```
Comment (shamsasari) [inline docs/source/api-index.rst]: This contains `ServiceInfo` and `ServiceType` which was moved here as part of the removal of advertised services. They're now only needed for configuring the notary (and the driver) but I imagine they'll eventually be removed. I suggest we move these to `net.corda.nodeapi.internal`.

File: docs/source/api-index.rst [hunk 3]
```diff
@@ -16,11 +16,34 @@ This section describes the APIs that are available for the development of CorDap
...
+Internal APIs and stability guarantees
+--------------------------------------
+
+As of Corda 1.0, the following modules export public API that we promise to maintain backwards compatibility for,
+unless an incompatible change is required for security reasons:
+
+* core
+* client-rpc
+* client-jackson
+* node-driver
```
Comment (shamsasari) [inline docs/source/api-index.rst]: I don't feel comfortable committing to backwards compatibly for the driver right now. Changes have gone into in this in recent weeks in a manner which shows that it's not been considered public API.

If we're committing to whatever we have now, then I suggest we do a passover first for anything obviously wrong. For example, `Driver.startNode` still has an `advertisedServices` parameter.

File: docs/source/api-index.rst (hunk 2 above)
Comment (shamsasari) [inline docs/source/api-index.rst]: `node-api` also has Artemis stuff and `VerifierApi` which shouldn't be public. There's also some config stuff. These will also need to be moved to internal.

Comment (mnesbit) [reply docs/source/api-index.rst]: Actually Artemis stuff will always have to be public, because that is the bus we use to communicate with the node as a client, although it is marked as internal.

File: docs/source/api-index.rst [hunk 4]
```diff
@@ -16,11 +16,34 @@ This section describes the APIs that are available for the development of CorDap
...
+* client-jfx, client-mock
 
 Code that falls into the following package namespaces are for internal use only and not public. In a future release the
 node will not load any CorDapp which uses them.
 
 * Any package in the ``net.corda`` namespace which contains ``.internal``
 * ``net.corda.node``
+
+For Corda 1.0 we do not currently provide a stable wire protocol or support for database upgrades. Therefore, you should
+not deploy 1.0 in production and you may need to do a database reset across upgrades until the necessary wire stability
```
Comment (gendal) [inline docs/source/api-index.rst]: "Therefore, you should not expect to be able to migrate persisted data from 1.0 to future versions."

(Point: we should not be telling people whether they should or should not deploy in production; that's a judgement and business call for them and we know several firms have already used Corda for production scenarios successfully - they had their eyes open and expectations appropriately set.)

**Comment by mikehearn** [comment]
OK - I think I've addressed the review comments. @Clintonio are you OK with stabilising the API for the Gradle plugins? If not, what hi-pri items need to be done?

**Comment by Clintonio** [comment]
@mike I wouldn't stablise the gradle plugins yet. I'll explain each any why now;

1) Publish utils - this is really useful but not at all documented or battle hardened. If we want to expose it as public we can but it's not really a part of any Corda offering - it's only there because it supports the other plugins and Corda itself
2) Quasar utils - this just needs a revisit. It did some nasty things before like adding a repo to your repos list automatically. I want either myself of Chris R to review it's purpose and usage.
3) Cordformation - this is the one we can consider API stablising. It's actually used actively and is genuinely Corda specific. One thing I want to do is to unify it with the Driver and other node runner/builders. I also want to split out the Cordapp building stuff if possible. While we prescribe the Cordapp format it's not necessary for it to be bundled with Cordformation. Conceptually these are distinct. NodeRunner may also needs a split.
4) Cordform common - this is a part of the unification between Corda builder/runners and Cordformation. I didn't create it initially but I did move it to this project. It's a part of Node and thus internal. It's not even a gradle plugin actually, just a dependency.

So tldr; only Cordformation can be stablised _but_ needs some planning. I'd suggest leaving it to a version shortly after 1 so I can have some time to plan it, write up the JIRAs and then hand them out to people (I don't necessarily need to do the work, just the design) and thinking.

**Comment by Clintonio** [comment]
If I was aware we were to stablise them (I've been telling people we explicitly aren't for ages) I'd have done the above already.

File: docs/source/api-index.rst [hunk 5]
```diff
@@ -16,11 +16,41 @@ This section describes the APIs that are available for the development of CorDap
...
+
+   Finally, please note that the 1.0 release has not yet been security audited.
+
+As of Corda 1.0, the following modules export public API that we promise to maintain backwards compatibility for,
+unless an incompatible change is required for security reasons:
+
+* core
+* client-rpc
+* client-jackson
+* node-api
```
Comment (shamsasari) [inline docs/source/api-index.rst]: With #1610 there's nothing in `node-api` that an app developer will need to depend on directly. We can drop this.

File: docs/source/api-index.rst [hunk 6]
```diff
@@ -16,11 +16,40 @@ This section describes the APIs that are available for the development of CorDap
...
+* The Gradle plugins: cordformation, quasar-utils
+
+The following modules don't yet have a completely stable API, but we will do our best to minimise disruption to
+developers using them until we are able to graduate them into the public API:
+
+* node-driver
+* confidential-identities
+* test-utils
+* client-jfx, client-mock
+* finance
```
Comment (mnesbit) [inline docs/source/api-index.rst]: IMHO Finance is likely to change a lot as we get clearer business logic demands. Currently it is just a show-case of what you can do with corda patterns.

File: docs/source/api-index.rst [hunk 7]
```diff
@@ -16,11 +16,40 @@ This section describes the APIs that are available for the development of CorDap
...
+* anything under the experimental directory
```
Comment (mnesbit) [inline docs/source/api-index.rst]: Should we be responsible at all for stability in experimental?

Comment (mikehearn) [reply docs/source/api-index.rst]: Nope, I added something in brackets to note that things in experimental may never graduate.

File: docs/source/api-index.rst (hunk 6 above)
Comment (mikehearn) [reply docs/source/api-index.rst]: Yeah, agreed.

File: docs/source/api-index.rst [hunk 8]
```diff
@@ -16,11 +16,40 @@ This section describes the APIs that are available for the development of CorDap
 
 Before reading this page, you should be familiar with the :doc:`key concepts of Corda <key-concepts>`.
 
-Internal
---------
+Internal APIs and stability guarantees
+--------------------------------------
+
+.. warning:: For Corda 1.0 we do not currently provide a stable wire protocol or support for database upgrades.
```
Comment (shamsasari) [inline docs/source/api-index.rst]: What about the JSON format produced from `client-jackson`?

**Comment by mikehearn** [comment]
OK, I rebased, squashed and I think this is good to go now. I added a comment about the json format produced by the jackson module.

**Comment by shamsasari** [comment]
I can't see the json format comment

**Comment by mikehearn** [comment]
Oops, forgot to re-push.
